app/js/lib/foxbox/services/ip-camera.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+
+import BaseService from './base';
+
+const TYPE = 'IpCamera';
+
+const p = Object.freeze({
+  // Public getter only properties.
+  manufacturer: Symbol('manufacturer'),
+  model: Symbol('model'),
+  name: Symbol('name'),
+
+  // Private getter only properties.
+  adapter: Symbol('adapter'),
+});
+
+export default class IpCameraService extends BaseService {
+  constructor(props, config) {
+    super(props, config);
+
+    // Public getter only properties.
+    this[p.manufacturer] = props.properties.manufacturer;
```

Comment:
I wasn't too sure whether it was going to be optional properties, but I think it makes a lot of sense to move them to the base class. I'll default them to an empty string.